browser/src/layer/tile/CanvasTileLayer.js

```diff
@@ -5428,68 +5456,114 @@ L.CanvasTileLayer = L.Layer.extend({
 			var centerTwips = this._pixelsToTwips(centerPos);
 
 			// Perform action
-			console.log('Automated User: Move to ',pos,' click at ', centerPos, centerTwips);
+			console.log('Randomize Settings: Move to ',pos,' click at ', centerPos, centerTwips);
 			this._map.fire('scrollto', pos);
 			this._postMouseEvent('buttondown', centerTwips.x, centerTwips.y, 1, 1, 0);
 			this._postMouseEvent('buttonup', centerTwips.x, centerTwips.y, 1, 1, 0);
 		}
 	},
 
-	_debugAutomatedUserInputTimeout: function () {
-		switch (this._debugAutomatedUserInputCase) {
-			case 0:
-				console.log('Automated User: Click somewhere visible');
-				var pos = this._pixelsToTwips(this._map._getCenterLayerPoint());
-				this._postMouseEvent('buttondown',pos.x,pos.y,1,1,0);
-				this._postMouseEvent('buttonup',pos.x,pos.y,1,1,0);
-				break;
-			case 1:
-				console.log('Automated User: Type text');
-				this._debugTypeText('asdf\nqwer\n', 100);
-				break;
-			case 2:
-				console.log('Automated User: Click formula bar');
-				this._map.sendUnoCommand('.uno:StartFormula');
-				break;
-			case 3:
-				console.log('Automated User: Type formula');
-				this._debugTypeText('A1\n', 100);
-				break;
-			case 4:
-				console.log('Automated User: Delete row');
-				this.postKeyboardEvent('input', 0, 1025); //up
-				this.postKeyboardEvent('input', 0, 1025); //up
-				this._map.sendUnoCommand('.uno:DeleteRows');
-				break;
-			case 5:
-				console.log('Automated User: Delete cells');
-				app.socket.sendMessage('removetextcontext id=0 before=0 after=1'); //delete
-				this.postKeyboardEvent('input', 0, 1025); //up
-				app.socket.sendMessage('removetextcontext id=0 before=0 after=1'); //delete
-				break;
-			case 6:
-				console.log('Automated User: Insert Shape');
-				var shapes = ['smiley','sun','moon','lightning','heart','flower'];
-				var shape = shapes[Math.floor(Math.random() * shapes.length)];
-				this._map.sendUnoCommand('.uno:SymbolShapes.'+shape);
-				// app.socket.sendMessage('removetextcontext id=0 before=0 after=1');
-				break;
-			case 7:
-				console.log('Automated User: Type in Shape');
-				this.postKeyboardEvent('input',0, 1280); // enter to select text
-				this._debugTypeText('textinshape', 100);
-				break;
-			case 8:
-				console.log('Automated User: Type Escape');
-				this.postKeyboardEvent('input',0, 1281); // esc
-				break;
-			default:
-				this._debugAutomatedUserInputCase = 0;
+	_debugAutomatedUserTypeCellFormula: function () {
+		setTimeout(L.bind(function () {
+			console.log('Automated User: Click somewhere visible');
+			var pos = this._pixelsToTwips(this._map._getCenterLayerPoint());
+			this._postMouseEvent('buttondown',pos.x,pos.y,1,1,0);
+			this._postMouseEvent('buttonup',pos.x,pos.y,1,1,0);
+		}, this), 0);
+
+		setTimeout(L.bind(function () {
+			console.log('Automated User: Type text');
+			this._debugTypeText('asdf\nqwer\n', 100);
```

Comment:
I'd want _debugTypeText hooked into the state machine logic really; typing big chunks of text as a ton of keystrokes in a single message, or as a single message with lots of keystrokes in it is a very unusual behavior; people type characters in one at a time - and we need that time to be realistic.
So the majority of _debugTypeText calls should send just 1 character I think.